### DISCUSSION TITLE
Weighted ANN Search

### DISCUSSION THREAD

**Discussion by jasonpump** [lancedb/lancedb#549] [General]
Hi Lancedb team,

Is that possible that we can encode the example with multiple embeddings and then when do ANN search, we can do something like weighted search? For example 0.1 * Similarity_Score_A + 0.9 * Similarity_Score_B.
Thanks!

**Comment by jasonpump** [comment]
ping, thanks!

**Reply by rationalga** [reply]
Hi, 
It looks like you want to use some sort of ensemble technique to combine results from different models. If two different models are used to create embedding then two separate indexes needs to be created. You can find the nearest neighbours from one index and similarly find candidates from the other index and use your weighted scheme to refine results. 
Thanks.
PS: I am not affiliated with Lancedb.
